libraries/aws/shadow/integration-test/shadow_system_test.c

```diff
@@ -0,0 +1,818 @@
+                                                      SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ),
+                                                      MQTTQoS0 ) );
+
+    /* Subscribe to shadow topic /update/documents with Qos 0. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic( &context,
+                                                      SHADOW_TOPIC_STRING_UPDATE_DOCUMENTS( THING_NAME ),
+                                                      SHADOW_TOPIC_LENGTH_UPDATE_DOCUMENTS( THING_NAME_LENGTH ),
+                                                      MQTTQoS0 ) );
+
+    /* First of all, try to delete any Shadow document in the cloud. 
+     * This could trigger the /delete/accepted or /delete/rejected 
+     * based on the thing status on the cloud.
+     */
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    SHADOW_TOPIC_STRING_DELETE( THING_NAME ),
+                                                    updateDocument,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /delete/accepted or /delete/rejected. */
+    TEST_ASSERT_TRUE( ( receivedDeleteAcceptedResult || receivedDeleteRejectedResult ) );
+
+    /* Publish to the shadow topic /update with reported payload, 
+    *  that we subscribed to, with Qos 0. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    SHADOW_TOPIC_STRING_UPDATE( THING_NAME ),
+                                                    TEST_SHADOW_DESIRED,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /update/documents*/
+    TEST_ASSERT_TRUE( receivedUpdateDocumentsResult );
+
+    /* Check the flag for /update/delta. */
+    TEST_ASSERT_TRUE( receivedUpdateDeltaResult );
+
+    /* Check the flag for /update/accepted and /update/rejected. */
+    TEST_ASSERT_TRUE( ( receivedUpdateAcceptedResult || receivedUpdateRejectedResult ) );
+
+    /* Finally, sending null payload on topic /get to trigger /get/accepted. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    SHADOW_TOPIC_STRING_GET( THING_NAME ),
+                                                    updateDocument,
+                                                    MQTTQoS0 ) );
+
+    /* Check the flag for /get/accepted and /get/rejected. */
+    TEST_ASSERT_TRUE( ( receivedGetAcceptedResult || receivedGetRejectedResult ) );
+
+    /* Un-subscribe from a topic with Qos 0. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, unsubscribeFromTopic( &context,
+                                                          SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
+                                                          SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ),
+                                                          MQTTQoS0 ) );
+
+    
+    
+}
```

Comment:
I think uncrustify should add newlines to source code?

Reply:
please review [this commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1134/commits/87cb11ae1496f618c50d49ad2ed7fd5c3630a3ec) for the update.

Reply:
There is still a new line here